tests/failpoints/cases/test_merge.rs

```diff
@@ -248,6 +248,49 @@ fn test_node_merge_catch_up_logs_restart() {
     must_get_equal(&cluster.get_engine(3), b"k11", b"v11");
 }
 
+/// Test if merge is working properly when merge entries is empty but commit index is not updated.
+#[test]
+fn test_node_merge_catch_up_logs_empty_entries() {
+    let _guard = crate::setup();
+    let mut cluster = new_node_cluster(0, 3);
+    configure_for_merge(&mut cluster);
+    cluster.run();
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k3", b"v3");
+
+    let pd_client = Arc::clone(&cluster.pd_client);
+    let region = pd_client.get_region(b"k1").unwrap();
+    let peer_on_store1 = find_peer(&region, 1).unwrap().to_owned();
+    cluster.must_transfer_leader(region.get_id(), peer_on_store1);
+    cluster.must_split(&region, b"k2");
+    let left = pd_client.get_region(b"k1").unwrap();
+    let right = pd_client.get_region(b"k2").unwrap();
+
+    // make sure the peer of left region on engine 3 has caught up logs.
+    cluster.must_put(b"k0", b"v0");
+    must_get_equal(&cluster.get_engine(3), b"k0", b"v0");
+
+    // let the source peer not apply the latest log
+    fail::cfg("before_set_applied_index_1000_1003", "return()").unwrap();
+    fail::cfg("apply_before_prepare_merge_1000_1003", "return()").unwrap();
+    pd_client.must_merge(left.get_id(), right.get_id());
+    thread::sleep(Duration::from_millis(100));
```

Comment:
It is not necessary to use failpoint to test this case. It should be sufficient to just construct a cluster with two peers merged, one peer has enough logs but still not committed, and start it to see if it can still be merged.